eos_data_distribution/avahi_monitor.py

```diff
@@ -79,23 +96,102 @@ def __init__(self):
         check_call(
             ["nfdc", "set-strategy", str(SUBSCRIPTIONS_SOMA), 'ndn:/localhost/nfd/strategy/multicast'])
 
-        self._registry = dict()
+        self.update_routed_status()
+        self.process_route_changes(False)
+
+    def update_routed_status(self):
+        self._routed = set(
+            self._ndn_gateways.values()).intersection(self.gateways)
+
+    def process_route_changes(self, prev_routed):
+        if self._routed:
+            logger.info(
+                "Being routed by an NDN node at %s (via %s), NOT acting as an edge router", address, ifname)
+            if not prev_routed:
+                # need to remove old links...
+                self._registery.values().map(self.remove_nexthop)
+                # ..and only add one to the routers
+                self._ndn_gateways.keys().map(self.add_nexthop)
+            else:
+                # no change: i am routed, i was routed before: no links
+                pass
+        else:
+            logger.info(
+                "Not Being routed by an NDN node, acting as an edge router")
+            if prev_routed:
+                # need to add back old links...
+                # this is limited to MAX_PEERS in add_nexthop
+                self._registry.values().map(self.add_nexthop)
+            else:
+                # nochange: wasn't routed, not routed now
+                pass
 
     def service_added_cb(self, sda, interface, protocol, name, type, h_type, domain, host, aprotocol, address, port, txt, flags):
-        ifname = sda.siocgifname(interface)
-        print "Found Service data for service '%s' of type '%s' (%s) in domain '%s' on %s.%i:" % (name, h_type, type, domain, ifname, protocol)
+        logger.debug(
```

Comment:
Changing from `print` to `logger.debug` should ideally be done in a separate commit.